internal/grpc_dialer.go

```diff
@@ -109,7 +111,18 @@ func dial(params dialParameters) (*grpc.ClientConn, error) {
 }
 
 func requiredInterceptors(metricScope tally.Scope, headersProvider HeadersProvider) []grpc.UnaryClientInterceptor {
-	interceptors := []grpc.UnaryClientInterceptor{metrics.NewScopeInterceptor(metricScope), errorInterceptor}
+	interceptors := []grpc.UnaryClientInterceptor{
+		errorInterceptor,
+		// Report aggregated metrics for the call, this is done outside of the retry loop.
+		metrics.NewRPCMetricsInterceptor(metricScope, "_total"),
+		// By default the grpc retry interceptor *is disabled*, preventing accidental use of retries.
+		// We add call options for retry configuration based on the values present in the context.
+		retry.NewRetryOptionsInterceptor(),
+		// Performs retries *IF* retry options are set for the call.
+		grpc_retry.UnaryClientInterceptor(),
+		// Report metrics for every call made to the server.
+		metrics.NewRPCMetricsInterceptor(metricScope, ""),
+	}
```

Comment:
This change in the interceptor chain is at the core of this PR.

Reply:
Would it be possible to create an interceptor that just combines these four post-error interceptors and use just that here? That would help ensure this ordering is maintained and provide a nice encapsulation.

Reply:
We could, but wouldn't it just be hiding details and ultimately moving the list from one place into another, where things can also be reordered?

Reply:
I suppose that's true, the benefit is just that it groups this stuff that "needs to go together", but, not a big deal either way